### PR TITLE
Update Android Gradle plugin to 4.1 version.

### PR DESCRIPTION
**Pull request by Tapchicoma**
Breaking change as I used new api and dropped support for build variants meta tasks. New api does not provide such information.

### CODE REVIEW

File: samples/kotlin-mpp-android/build.gradle.kts [hunk 1]
```diff
@@ -8,17 +8,8 @@ kotlin {
...
-            }
-        }
-
-        val androidMain by getting {
-            dependencies {
-                implementation(kotlin("stdlib"))
-            }
-        }
+        val commonMain by getting
+        val androidMain by getting
```
Comment (JLLeitschuh) [inline samples/kotlin-mpp-android/build.gradle.kts:12]: Are these still needed at all?

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/android/AndroidPluginsApplier.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
+@file:Suppress("UnstableApiUsage")
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/android/AndroidPluginsApplier.kt:1]: Can you apply this a bit more specifically?

**Review by JLLeitschuh — CHANGES_REQUESTED** [review]
A few small things.

**Comment by Tapchicoma** [comment]
@JLLeitschuh sorry for not addressing your review for a long time. Just did it, please check fixes.

**Review by JLLeitschuh — APPROVED** [review]
LGTM Thanks!

**Pull request merged**